dali/pipeline/data/buffer.h

```diff
@@ -399,7 +399,7 @@ class DLL_PUBLIC Buffer {
 
   TypeInfo type_ = {};               // Data type of underlying storage
   shared_ptr<void> data_ = nullptr;  // Pointer to underlying storage
-  AllocFunc allocate_;               // Custom allocation function
+  AllocFunc allocate_ = {};          // Custom allocation function
```

Comment:
This is actually a false positive. `AllocFunc` is a typedef for `std::function` and it has a well-defined default constructor.
https://en.cppreference.com/w/cpp/utility/functional/function/function